File: c/Sorting Algorithms/selectionSort.c
/*selection sort*/
#include <stdio.h>

void selectionSort(int[], int);
int main()
{
    int a[100], i, n;

    printf("Enter size of the  array : ");
    scanf("%d", &n);
    printf("Enter elements in array : \n");
    for (i = 0; i < n; i++)
        scanf("%d", &a[i]);

    selectionSort(a, n);

    printf("The sorted Array : \n");
    for (i = 0; i < n; i++)
        printf("%d\t", a[i]);
    printf("\n");
    return 0;
}

void selectionSort(int a[], int n)
{
    int i, j, temp, min;
    for (i = 0; i < n; i++)
    {
        min = i;
        for (j = i + 1; j < n; j++)
            if (a[j] < a[min])
                min = j;

        if (min != i)
        {
            temp = a[min];
            a[min] = a[i];
            a[i] = temp;
        }
    }
}

File: JavaScript/Sorting/SelectionSort/selection_sort.js
function selection_sort(inputArray){

    for(let i = 0; i<inputArray.length;i++){
        let min = i;
        for(let j = i+1; j<inputArray.length;j++){
            if(inputArray[min] > inputArray[j]){
                min = j;
            }
        }
        if( min != i){
            let temp = inputArray[i];
            inputArray[i] = inputArray[min];
            inputArray[min] = temp;
        }
    }
    return inputArray;
}

var nums = [6, 3, 4, 5, 61, 2, 77];
console.log(selection_sort(nums));
